home/tests/test_api.py

```diff
@@ -244,118 +254,142 @@ def test_pagination(self):
             )
             body.append(("Whatsapp_Message", block_value))
 
-        self.content_page1.whatsapp_body = body
-        self.content_page1.save_revision().publish()
+        page1.whatsapp_body = body
+        page1.save_revision().publish()
 
         # it should only return the 11th paragraph if 11th message
         # is requested
-        response = self.client.get(
-            f"/api/v2/pages/{self.content_page1.id}/?whatsapp=True&message=11"
-        )
+        response = uclient.get(f"/api/v2/pages/{page1.id}/?whatsapp=True&message=11")
         content = json.loads(response.content)
-        self.assertEqual(content["body"]["message"], 11)
-        self.assertEqual(content["body"]["next_message"], 12)
-        self.assertEqual(content["body"]["previous_message"], 10)
-        self.assertEqual(content["body"]["text"]["value"]["message"], "WA Message 11")
+        assert content["body"]["message"] == 11
+        assert content["body"]["next_message"] == 12
+        assert content["body"]["previous_message"] == 10
+        assert content["body"]["text"]["value"]["message"] == "WA Message 11"
 
-    def test_number_of_queries(self):
-        with self.assertNumQueries(8):
-            self.client.get("/api/v2/pages/")
+    def test_number_of_queries(self, uclient, django_assert_num_queries):
+        """
+        FIXME:
+         * Is the number of queries significant? What does it depend on?
+         * Should this be part of some other test?
+        """
+        with django_assert_num_queries(8):
+            uclient.get("/api/v2/pages/")
 
-    def test_detail_view(self):
+    def test_detail_view(self, uclient):
+        """
+        FIXME:
+         * It's unclear what this is actually testing.
+         * Should it be multiple tests instead of just one?
+         * This should probably be in a class without the content setup.
+        """
```

Comment:
I think this is testing the format of the `/pages/{page_id}/` endpoint. I think this does need to be multiple tests, since there's a bit of `?whatsapp=true` in there, which is a bit odd, that bit can be a separate test, I think here we just want to test the bits of the response that are the same across all channel types. It also tests that page views are created on a `GET` to the page.

We could have it as a separate class, or use one of the ContentPages we've already created in the setup, instead of deleting and creating a new one.

Reply:
🤔

Reply:
I've split this into four separate tests with more thorough assertions that use the existing page data instead of replacing it with new data.